server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSReadOptionsCheckUtility.java

```diff
@@ -0,0 +1,98 @@
+
+/** Checks CICS READ rules for required and invalid options */
+public class CICSReadOptionsCheckUtility extends CICSOptionsCheckBaseUtility {
+
+    public static final int RULE_INDEX = RULE_cics_read;
+
+    private static final Map<Integer, ErrorSeverity> DUPLICATE_CHECK_OPTIONS =
+        new HashMap<Integer, ErrorSeverity>() {
+            {
+                put(CICSLexer.FILE, ErrorSeverity.ERROR);
+                put(CICSLexer.UNCOMMITTED, ErrorSeverity.ERROR);
+                put(CICSLexer.CONSISTENT, ErrorSeverity.ERROR);
+                put(CICSLexer.REPEATABLE, ErrorSeverity.ERROR);
+                put(CICSLexer.UPDATE, ErrorSeverity.ERROR);
+                put(CICSLexer.TOKEN, ErrorSeverity.ERROR);
+                put(CICSLexer.INTO, ErrorSeverity.ERROR);
+                put(CICSLexer.SET, ErrorSeverity.ERROR);
+                put(CICSLexer.RIDFLD, ErrorSeverity.ERROR);
+                put(CICSLexer.KEYLENGTH, ErrorSeverity.ERROR);
+                put(CICSLexer.GENERIC, ErrorSeverity.ERROR);
+                put(CICSLexer.SYSID, ErrorSeverity.ERROR);
+                put(CICSLexer.LENGTH, ErrorSeverity.ERROR);
+                put(CICSLexer.DEBKEY, ErrorSeverity.ERROR);
+                put(CICSLexer.DEBREC, ErrorSeverity.ERROR);
+                put(CICSLexer.RBA, ErrorSeverity.ERROR);
+                put(CICSLexer.RRN, ErrorSeverity.ERROR);
+                put(CICSLexer.XRBA, ErrorSeverity.ERROR);
+                put(CICSLexer.EQUAL, ErrorSeverity.ERROR);
+                put(CICSLexer.GTEQ, ErrorSeverity.ERROR);
+                put(CICSLexer.NOSUSPEND, ErrorSeverity.ERROR);
+            }
+        };
+
+    public CICSReadOptionsCheckUtility(DialectProcessingContext context, List<SyntaxError> errors) {
+        super(context, errors, DUPLICATE_CHECK_OPTIONS);
+    }
+
+    /**
+     * Entrypoint to check CICS READ rules for required and invalid options
+     * @param ctx ParserRuleContext subclass containing options
+     * @param <E> A subclass of ParserRuleContext
+     */
+    public <E extends ParserRuleContext> void checkOptions(E ctx) {
+        if (ctx.getRuleIndex() == CICSParser.RULE_cics_read_body)
+            checkRule((CICSParser.Cics_read_bodyContext) ctx);
+    }
+
+    private void checkRule(CICSParser.Cics_read_bodyContext ctx) {
+        if (ctx.DATASET().isEmpty()) {
+            checkHasMandatoryOptions(ctx.FILE(), ctx, "FILE");
+        }
+
+        checkMutuallyExclusiveOptions("FILE instead of DATASET", ctx.FILE(), ctx.DATASET());
+
+        checkHasMandatoryOptions(ctx.RIDFLD(), ctx, "RIDFLD");
+
+        checkMutuallyExclusiveOptions("UNCOMMITTED, CONSISTENT, REPEATABLE or UPDATE", ctx.UNCOMMITTED(), ctx.CONSISTENT(), ctx.REPEATABLE(), ctx.UPDATE());
+
+        checkPrerequisiteIsMet(ctx.UPDATE(), ctx.TOKEN(), ctx, "TOKEN without UPDATE");
+
+        checkMutuallyExclusiveOptions("INTO or SET", ctx.INTO(), ctx.SET());
```

Comment:
```suggestion
        checkHasExactlyOneOption("INTO or SET", ctx, ctx.INTO(), ctx.SET());
```
Into or Set is mandatory